src/app/space/wizard/space-wizard.component.ts

```diff
@@ -53,39 +53,47 @@ export class SpaceWizardComponent implements OnInit, OnDestroy {
    * by invoking the spaceService
    */
   createSpace() {
-    console.log('Creating space', this.space);
+    if (!this.userService.currentLoggedInUser && !this.userService.currentLoggedInUser.id) {
+      console.log('Error creating space, invalid user.', this.userService.currentLoggedInUser);
+      this.notifications.message(<Notification> {
+        message: `Failed to create "${this.space.name}". Invalid user: "${this.userService.currentLoggedInUser}"`,
+        type: NotificationType.DANGER
+      });
+      return;
+    }
+
+    console.log('Creating space', this.space, this.userService.currentLoggedInUser.id);
     if (!this.space) {
       this.space = this.createTransientSpace();
     }
     this.space.attributes.name = this.space.name.replace(/ /g, '_');
-    this.userService.getUser()
-      .switchMap(user => {
-        this.space.relationships['owned-by'].data.id = user.id;
-        return this.spaceService.create(this.space);
-      })
-      .do(createdSpace => {
-        this.spacesService.addRecent.next(createdSpace);
-      })
-      .switchMap(createdSpace => {
-        return this.spaceNamespaceService
-          .updateConfigMap(Observable.of(createdSpace))
-          .map(() => createdSpace)
-          // Ignore any errors coming out here, we've logged and notified them earlier
-          .catch(err => Observable.of(createdSpace));
-      })
-      .subscribe(createdSpace => {
-          this.router.navigate([createdSpace.relationalData.creator.attributes.username,
-            createdSpace.attributes.name]);
-          this.finish();
-        },
-        err => {
-          console.log('Error creating space', err);
-          this.notifications.message(<Notification> {
-            message: `Failed to create "${this.space.name}"`,
-            type: NotificationType.DANGER
-          });
-          this.finish();
+
+    this.space.relationships['owned-by'].data.id = this.userService.currentLoggedInUser.id;
+    this.spaceService.create(this.space)
+    .first()
```

Comment:
Is it confirmed that the returned `Observable<Space>` from `spaceService.create` is a hot Observable? I don't know the implementation details but it seems like that's the kind of thing that would just emit once and complete - similar to what a `Promise` would've been used for. Our discussion about this before was on using `.first()` or a non-empty filter on the userService so that the logged-out empty user wouldn't trigger this code path a second time.

You can verify it easily enough by adding a `.do(noop, noop, () => console.log('subscription completed'))` to the `spaceService.create`, so that you get a browser console message when the Observable completes (if ever). If that Observable already completes after emitting a Space then there's no need for the `.first()` operator. If it doesn't complete, then that's pretty weird, but the `.first()` operator would be reasonable.

Reply:
Yeah, looks like it does complete. Does this mean I don't need to unsubscribe?

Reply:
Yea, no need to explicitly unsubscribe here then. Nice.